src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,119 @@
+
+    public List<Task> getTasks() {
+        archiveTasks();
+        return taskRepository.findAll();
+    }
+
+    public List<Task> getAllLoggedUserTasks(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUser(loggedUser);
+    }
+
+    public List<Task> getAllLoggedUserTasksOrderByImportance(Principal principal) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserOrderByImportance(loggedUser);
+    }
+
+    public List<Task> getLoggedUserParticularTasks(Principal principal, Importance importance) {
+        User loggedUser = userRepository.findByLogin(principal.getName()).orElseThrow(()
+                -> new EntityNotFoundException("User not found"));
+        archiveTasks();
+        return taskRepository.findAllByUserIdaAndImportance(loggedUser.getId(), importance);
+    }
+
+    public List<Task> getUserTasks(Long userId) {
+        archiveTasks();
+        return taskRepository.findALLByUserId(userId);
+    }
+
+    public Task getTaskById(Long taskId) {
+        archiveTasks();
+        Optional<Task> task = taskRepository.findById(taskId);
+        return task.get();
+    }
+
+    public Task addTask(Task task, Principal principal) {
+        task.setImportance(DEFAULT_IMPORTANCE);
+        task.setCreationDate(LocalDateTime.now());
+        task.setExpirationDate(endOfTheDay());
+        task.setUser(userRepository.findByLogin(principal.getName()).get());
+        return taskRepository.save(task);
+    }
+
+    public Task updateTask(Task task, Long id) {
+        Task updatedTask = taskRepository.findById(id).get();
+        updatedTask.setImportance(task.getImportance());
+        updatedTask.setDescription(task.getDescription());
+        updatedTask.setArchived(task.isArchived());
+        updatedTask.setDone(task.isDone());
+        return taskRepository.save(updatedTask);
+    }
+
+    public void deleteTask(Long id) {
+        taskRepository.deleteById(id);
+    }
+
+
+    public List<Task> getTaskByImportance(Importance importance) {
+        return taskRepository.findAllByImportance(importance);
+    }
+
+
+    public LocalDateTime endOfTheDay() {
+        LocalDateTime now = LocalDateTime.now();
+        return LocalDateTime.of(LocalDate.now(), LocalTime.of(23, 59, 59));
+    }
+
+    public void archiveTasks() {
+        List<Task> tasks = taskRepository.findAll();
+        tasks.forEach(task -> {
+            if (LocalDateTime.now().isAfter(task.getExpirationDate())) {
+                task.setArchived(true);
+                taskRepository.save(task);
+            }
+        });
+    }
+
+    public Optional<Long> getSumOfArchivedTasksById(Long userId) {
```

Comment:
What is the difference between this method and the one below?

Reply:
This one is for admin, second for logged user.

Reply:
The both do the same though. Remove one or convince me why we should leave both of them.